include/pms-utils/vdb/vdb.hpp

```diff
@@ -0,0 +1,203 @@
+#pragma once
+
+#include "pms-utils/atom/atom.hpp"
+#include "pms-utils/depend/depend.hpp"
+#include "pms-utils/ebuild/ebuild.hpp"
+#include "pms-utils/misc/macro-begin.hpp"
+
+#include <bits/fs_dir.h>
+#include <cstddef>
+#include <cstdint>
+#include <filesystem>
+#include <optional>
+#include <variant>
+#include <vector>
+
+namespace [[gnu::visibility("default")]] pms_utils {
+namespace vdb {
+
+struct Obj {
+    std::filesystem::path path;
+    std::string md5;
+    std::uint64_t size;
+};
+
+struct Dir {
+    std::filesystem::path path;
+};
+
+struct Sym {
+    std::filesystem::path src;
+    std::filesystem::path dest;
+    std::uint64_t size;
+};
+
+using Content = std::variant<Obj, Dir, Sym>;
+
+struct Package {
+    pms_utils::atom::Name name;
+    pms_utils::atom::Version version;
+};
+
+class Entry {
+public:
+    Entry(std::filesystem::path);
```

Comment:
nodiscard, parameter name?

should be `[[nodiscard]] Entry(std::filesystem::path path)`

also please put this in the second public block for consistency with the other classes